remoting/src/main/java/org/springframework/security/remoting/dns/JndiDnsResolver.java

```diff
@@ -107,15 +120,18 @@ private String resolveServiceEntry(String serviceType, String domain, DirContext
 				int priority = Integer.parseInt(record[0]);
 				int weight = Integer.parseInt(record[1]);
 				// we have a new highest Priority, so forget also the highest weight
+				int SERVICE_RECORD_PORT_INDEX = 2;
```

Comment:
Since we're already using the integers directly for the priority, weight and hostname, I think we should continue that pattern here and use `record[2]` rather than setting a constant.